cpp/ql/src/Security/CWE/CWE-129/ImproperArrayIndexValidation.ql

```diff
@@ -123,5 +123,5 @@ where
   conf.hasFlowPath(source, sink) and
   isFlowSource(source.getNode(), sourceType)
 select sink.getNode(), source, sink,
-  "$@ flows to here and is used in an array indexing expression, potentially causing an invalid access.",
+  "An array indexing expression depends on $@, potentially causing an invalid access.",
```

Comment:
I'm not sure about "depends on" here.  I'm not sure I like the original message very much either.  Neither seems to get across the idea that the problem is that it hasn't been bounds checked.

Reply:
I'm using "depends on" because it's a taint-tracking query. "flows to" sounds like the value is unmodified between the source and the sink.  

I'll try to get across that the value hasn't been bounds checked.

Reply:
I think because we're blocking flow at bounds checks, it feels somewhere between a taint flow and a data flow query.

Reply:
Happy with your latest version. :+1: